internal/pgengine/bootstrap.go

```diff
@@ -173,6 +173,23 @@ func (pge *PgEngine) getPgxConnConfig() *pgxpool.Config {
 		pgconn.TypeMap().RegisterType(&pgtype.Type{Name: "timetable.log_type", OID: pge.logTypeOID, Codec: &pgtype.EnumCodec{}})
 		return err
 	}
+	// reset session before returning connection to the pool
+	// after task completion, if the task is not properly finalized (especially when running in autonomous mode), some objects and/or setting changes will still exist in the session
+	// if this session is then reused it can cause task failure / unintended side effects
+	connConfig.AfterRelease = func(pgconn *pgx.Conn) bool {
+
+		batch := pgx.Batch{}
+		batch.Queue("CLOSE ALL;")
+		batch.Queue("SET SESSION AUTHORIZATION DEFAULT;")
+		batch.Queue("RESET ALL;")
+		batch.Queue("SELECT pg_advisory_unlock_all();")
+		batch.Queue("DISCARD TEMP;")
+		batch.Queue("UNLISTEN *;")
```

Comment:
Not sure we want this since every connection should `LISTEN <clientname>` during it's lifetime.

Reply:
I wasn't  100%sure about that one. 
In that case we need to query pg_listening_channels ()  and unlisten all channels pg_timetable isn't directly subscribed to.